### PR TITLE
ci: let dependabot update used GitHub action's versions

### PR DESCRIPTION
**Pull request by consideRatio**
It seems that dependabot can keep our workflows used GitHub actions' versions, so I figure why not!

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+  # Maintain Python dependencies for the jupyterhub/k8s-hub image
+  - package-ecosystem: pip
+    directory: "/images/hub"
+    schedule:
+      interval: daily
+    open-pull-requests-limit: 4
+    versioning-strategy: lockfile-only
+    labels:
+      - maintenance
+      - dependencies
```
Comment (manics) [inline .github/dependabot.yaml:13]: This is overriding the default dependabot labels:
https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/customizing-dependency-updates#setting-custom-labels

Are we using them for a particular purpose, or could we remove this bit of the config and rely on the defaults?

Comment (consideRatio) [reply .github/dependabot.yaml:13]: Interesting to know! Hmm want github-activity to sort these PR quickly when generating a changelog, which based on current github-activity function means the maintenance label.

Id like a dedicated category for dependencies etc, but at this point its maintenance.

Can we add extra labels in a hardcoded way instead?

Comment (manics) [reply .github/dependabot.yaml:13]: Sure, that's fine 😀. Just wanted to check it's an intentional override.

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+
+  # Maintain dependencies in our GitHub Workflows
+  - package-ecosystem: "github-actions"
+    directory: "/"  # This should be / rather than .github/workflows
+    schedule:
+      interval: daily
+    labels:
+      - maintenance
+      - dependencies
```
Comment (consideRatio) [inline .github/dependabot.yaml:22]:
```suggestion
      - dependencies

```

File: .github/dependabot.yaml (hunk 1 above)
Comment (consideRatio) [reply .github/dependabot.yaml:13]: :) thanks for review @manics! Merge?

**Review by manics — APPROVED** [review]

**Pull request merged**